src/etherscan.js

```diff
@@ -1,43 +1,87 @@
+import rp from 'request-promise';
 import fs from 'fs';
+import path from 'path';
+import { getAccessToken } from './config';
+
+
+const expectedResponse = {
+  OK: 'OK',
+};
+
 /**
- * 
- * This is private function 
- * that saves the file ABI object into file locally
+ *  Validate if Etherscan response was succssefful should
+ *  throw if a reponse had an error
+ *  @param reponse: object
  *
- * @param address
- * return: string 
  */
-export const saveABIFile = data => 'FILE_PATH'; // eslint-disable-line no-unused-vars
-
+const validateResponse = (response) => {
+  if (response.status && response.message === expectedResponse.OK) {
+    return;
+  }
+  throw new Error('Wrong response from Etherscan or wrong Contract Address');
+};
 /**
- * 
- * Get the ABI from `contracts` folder if eixsts otherwise get it 
- * from etherscan api then save it in file and return it 
+ *   Save Given ABI object in a local file
+ *   throw if a reponse had an error
+ *   @param abi: Object
+ *   @param filepath: string
  *
- * @param address
- * return: array 
  */
-export const getABI = async (address) => {
-  const filePath = `../contracts/${address}`;
-  let abi = null;
-  if (!fs.existsSync(filePath)) {
-    // @todo: request etherscan api
-
-    saveABIFile(abi);
-    abi = [];
-  } else {
-    abi = fs.readFileSync(filePath, { encoding: 'utf8' });
+const saveABIFile = async (abi, filepath) => {
+  try {
+    fs.writeFileSync(filepath, abi);
+  } catch (err) {
+    throw new Error('Problem with writing ABI file');
```

Comment:
Can you please also return the original error message `err.message`?

Reply:
What is the different between this implementation and the following? 
```
const saveABIFile = async (abi, filepath) => fs.writeFileSync(filepath, abi);
```

Reply:
You have a point